nav2_waypoint_follower/src/waypoint_follower.cpp

```diff
@@ -268,8 +270,14 @@ WaypointFollower::followWaypoints()
           get_logger(), "Completed all %zu waypoints requested.",
           goal->poses.size());
         result->missed_waypoints = failed_ids_;
-        action_server_->succeeded_current(result);
+        if (error_code_ > 0) {
```

Comment:
Woof, I forgot about this. WPF can have multiple failed requests. I think similar to how we handle `missed_waypoints` we need to handle the error codes

Reply:
Do we want the error codes for each of the failures or only when the action fails? 

I assumed that the old error codes do not provide any useful information to higher level autonomy systems and thus we only should return the last error. However, switching it should be quite simple.

Reply:
That's a really great question. I think it would be helpful context to a user to know not only why the last one failed, but for any that failed, why that occurred. It could be useful context for analysis of failure modes.

Reply:
It might actually be sensible then to make the Missed WP stuff a new struct to store in the WPF to keep that data associated in case we want to add more fields to that in the future `struct MissedWaypoints {int index; int error_code;}`.

Perhaps also updating the message as well for that kind of organization? If we did that, it might be sensible to also add the index's actual goal pose too.

Reply:
Could we create a new message type `MissedWaypoint.msg` and have the `FollowWaypoint.action` return a array of `MissedWaypoint.msg`?

Reply:
Yup!

Reply:
@jwallace42 was this dropped for some technical reason?